zeebe/journal/src/main/java/io/camunda/zeebe/journal/file/FlushableSegment.java

```diff
@@ -22,7 +24,8 @@ interface FlushableSegment {
    * <p>If the method returns true, then it is guaranteed that the modified pages for this segment *
    * have been flushed to disk (according to the underlying file system).
    *
+   * @throws FlushException if for any reason the flush failed
    * @return true if the segment flushed successfully, false otherwise
    */
-  boolean flush();
+  boolean flush() throws FlushException;
```

Comment:
:question: I wonder if the boolean return value makes sense anymore, since we now throw if the flush failed :shrug: 

:wrench: It probably doesn't, and we should change that. Can gladly be a follow up PR/issue.